src/app.rs

```diff
@@ -116,114 +110,12 @@ impl Yippee {
-                        self.events
-                            .push(EmbedderEvent::MouseWindowEventClass(event));
-                    }
-                }
-                WindowEvent::TouchpadMagnify { delta, .. } => {
-                    self.events.push(EmbedderEvent::Zoom(1.0 + delta as f32));
-                }
-                WindowEvent::MouseWheel { delta, phase, .. } => {
-                    // FIXME: Pixels per line, should be configurable (from browser setting?) and vary by zoom level.
-                    const LINE_HEIGHT: f32 = 38.0;
-
-                    let (mut x, mut y, mode) = match delta {
-                        winit::event::MouseScrollDelta::LineDelta(x, y) => {
-                            (x as f64, (y * LINE_HEIGHT) as f64, WheelMode::DeltaLine)
-                        }
-                        winit::event::MouseScrollDelta::PixelDelta(position) => {
-                            let position =
-                                position.to_logical::<f64>(self.webview.window.scale_factor());
-                            (position.x, position.y, WheelMode::DeltaPixel)
-                        }
-                    };
-
-                    // Wheel Event
-                    self.events.push(EmbedderEvent::Wheel(
-                        WheelDelta { x, y, z: 0.0, mode },
-                        DevicePoint::new(
-                            self.mouse_position.x as f32,
-                            self.mouse_position.y as f32,
-                        ),
-                    ));
-
-                    // Scroll Event
-                    // Do one axis at a time.
-                    if y.abs() >= x.abs() {
-                        x = 0.0;
-                    } else {
-                        y = 0.0;
-                    }
-
-                    let phase: TouchEventType = match phase {
-                        TouchPhase::Started => TouchEventType::Down,
-                        TouchPhase::Moved => TouchEventType::Move,
-                        TouchPhase::Ended => TouchEventType::Up,
-                        TouchPhase::Cancelled => TouchEventType::Cancel,
-                    };
-
-                    self.events.push(EmbedderEvent::Scroll(
-                        ScrollLocation::Delta(LayoutVector2D::new(x as f32, y as f32)),
-                        DeviceIntPoint::new(
-                            self.mouse_position.x as i32,
-                            self.mouse_position.y as i32,
-                        ),
-                        phase,
-                    ));
-                }
-                WindowEvent::CloseRequested => {
-                    self.events.push(EmbedderEvent::Quit);
-                }
-                e => log::warn!("Yippee hasn't supported this window event yet: {e:?}"),
-            },
+            } => self.webview.handle_winit_window_event(
+                &mut self.servo,
+                &mut self.events,
+                &mut self.mouse_position,
```

Comment:
Can you also move mouse position to webview module?

Reply:
pushed